basedecorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class Basedecorator < Nameable
+  def initialize(name)
+    super()
+    @name = name
+  end
+
```

Comment:
- [ **OPTIOANL** ] As required for the project, kindly ensure that the variable name you used is the same with the one mentioned in the requirements. I recommended you also apply this approach in other classes such as `Trimmerdecorator` and `Capitalizedecorator` 
![Screenshot from 2023-10-11 15-08-15](https://github.com/fatmahussein/school-library/assets/83042742/248c70bc-d22c-47be-ab37-24f4fecd9f4a)


- [ **OPTIONAL** ] I think it would be better if `Setter` and `Getter` are present on the `@nameble` instance variable. Using getter and setter methods allow to access and modify the value of a variable by providing a controlled and predictable interface for other objects to interact with the variable, while still maintaining the principles of encapsulation and data hiding. i.e. `attr_accessor :nameable`